store/tikv/region_cache.go

```diff
@@ -1671,42 +1890,99 @@ func (s *Store) requestLiveness(bo *Backoffer) (l livenessState) {
 	return
 }
 
-func invokeKVStatusAPI(saddr string, timeout time.Duration) (l livenessState) {
+func invokeKVStatusAPI(addr string, timeout time.Duration) (l livenessState) {
 	start := time.Now()
 	defer func() {
 		if l == reachable {
 			metrics.StatusCountWithOK.Inc()
 		} else {
 			metrics.StatusCountWithError.Inc()
 		}
-		metrics.TiKVStatusDuration.WithLabelValues(saddr).Observe(time.Since(start).Seconds())
+		metrics.TiKVStatusDuration.WithLabelValues(addr).Observe(time.Since(start).Seconds())
 	}()
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	url := fmt.Sprintf("%s://%s/status", config.InternalHTTPSchema(), saddr)
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+
+	conn, cli, err := createKVHealthClient(ctx, addr)
 	if err != nil {
-		logutil.BgLogger().Info("[liveness] build kv status request fail", zap.String("store", saddr), zap.Error(err))
+		logutil.BgLogger().Info("[health check] create grpc connection failed", zap.String("store", addr), zap.Error(err))
 		l = unreachable
 		return
 	}
-	resp, err := config.InternalHTTPClient().Do(req)
+	defer func() {
+		err := conn.Close()
+		if err != nil {
+			logutil.BgLogger().Info("[health check] failed to close the grpc connection for health check", zap.String("store", addr), zap.Error(err))
+		}
+	}()
+
+	req := &healthpb.HealthCheckRequest{}
```

Comment:
This one introduced a protocol change, what happens when a new versioned TiDB works with an old versioned TiKV?

Reply:
An unhealthy store will never become healthy. But is this situation common where new TiDB runs together with old version TiKV?

Reply:
It may affect the rolling update.
Seems not a big problem.

Reply:
Nope. In the normal process of upgrading, PD's version >= TiKV's version >= TiDB's version.